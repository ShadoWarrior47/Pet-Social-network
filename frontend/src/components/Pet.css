/* General Pet container styling */
.Pet {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    border: 1px solid #ddd;
    border-radius: 8px;
    padding: 10px;
    margin: 10px 0;
    background-color: #f9f9f9;
    box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
    transition: box-shadow 0.3s ease;
}

.Pet:hover {
    box-shadow: 0 4px 10px rgba(0, 0, 0, 0.2);
}

.Pet h3 {
    font-size: 1.5rem;
    margin: 8px 0px;
    color: #333;
}

.Pet p {
    margin: 4px 0;
    font-size: 1rem;
    color: #555;
}

.Pet img {
    width: 300px;
}