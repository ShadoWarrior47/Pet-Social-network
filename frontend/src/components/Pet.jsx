import React from "react";
import { Link } from "react-router-dom";
import "./Pet.css";

function Pet({ pet, onDelete, isMyPet = false }) {

    return (
        
            <div className="Pet">
                <h3><Link to={`/pets/${pet.id}`}>{pet.name}</Link></h3>
                <img src={pet.photoUrl} alt={pet.name} />
                
                {/* <p>{pet.type}</p>
            <p>Breed: {pet.breed}</p>
            <p>{pet.age} years old</p>
            <p>{pet.bio}</p> */}

            {isMyPet && (
                <>
                    <button>
                        <Link to={`/pets/edit/${pet.id}`}>Edit</Link>
                    </button>
                    <button onClick={() => onDelete(pet.id)}>Delete</button>
                </>
            )}
            </div>


    );
}

export default Pet;