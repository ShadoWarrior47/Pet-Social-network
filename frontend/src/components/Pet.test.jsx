import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Pet from "./Pet";
import '@testing-library/jest-dom';

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <Pet pet={{ id: 5, name: "Test Pet", type: "Dog" }} />
        </MemoryRouter>
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <Pet pet={{ id: 5, name: "Test Pet", type: "Dog" }} />
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("displays pet details correctly", function () {
    const { getByText } = render(
        <MemoryRouter>
            <Pet pet={{ id: 5, name: "Test Pet", type: "Dog" }} />
        </MemoryRouter>
    );

    expect(getByText("Test Pet")).toBeInTheDocument();
    // expect(getByText("This is a test company.")).toBeInTheDocument();
});
