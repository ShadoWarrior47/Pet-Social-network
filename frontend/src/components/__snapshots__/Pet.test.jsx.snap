// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Pet"
  >
    <h3>
      <a
        data-discover="true"
        href="/pets/5"
      >
        Test Pet
      </a>
    </h3>
    <img
      alt="Test Pet"
    />
  </div>
</DocumentFragment>
`;
